lib/ansible/modules/net_tools/basics/get_url.py

```diff
@@ -353,9 +353,9 @@ def url_get(module, url, dest, use_proxy, last_mod_time, force, timeout=10, head
     if info['status'] == 304:
         module.exit_json(url=url, dest=dest, changed=False, msg=info.get('msg', ''), elapsed=elapsed)
 
-    # Exceptions in fetch_url may result in a status -1, the ensures a proper error to the user in all cases
+    # NOTE: Exceptions in fetch_url may result in a status -1, the ensures a proper error to the user in all cases
     if info['status'] == -1:
-        module.fail_json(msg=info['msg'], url=url, dest=dest, elapsed=elapsed)
+        module.fail_json(elapsed=elapsed, **info)
```

Comment:
We also need to include `dest` here  in order to maintain backwards compatibility since it is not present in `info.
```suggestion
        module.fail_json(elapsed=elapsed,  dest=dest, **info)
```

Reply:
Key dest is already part of what is returned in info, so this will actually break the module.
You will get a "Duplicate key" error.

Reply:
https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/urls.py#L1312-L1323

Reply:
I'm sorry, but I still don't see `dest` in the output after this  change. What am I missing?

Before this patch:

```
TASK [get_url] ******************************************************************************
fatal: [localhost]: FAILED! => changed=false
  dest: /Users/sdoran/Downloads/file.png
  elapsed: 0
  msg: 'Request failed: <urlopen error unknown url type: htt>'
  state: absent
  url: htt://acme.com

After this patch:
```
fatal: [localhost]: FAILED! => changed=false
  elapsed: 0
  msg: 'Request failed: <urlopen error unknown url type: htt>'
  status: -1
  url: htt://samdoran.com/media/2017/keychain_access_vault_search.png
```

Reply:
I could be testing incorrectly, too. I just want to make sure we're _adding_ fields, not accidentally removing fields when we're adding new ones. Not trying to be a pain.

Reply:
You are right, and our integration tests are not sufficient in this regard. I'll fix.